core/server/master/src/test/java/alluxio/master/metrics/MetricsMasterTest.java

```diff
@@ -86,14 +93,14 @@ public void testAggregator() {
         Metric.from("worker.192_1_1_2.metricA", 1, MetricType.GAUGE),
         Metric.from("worker.192_1_1_2.metricB", 2, MetricType.GAUGE));
     mMetricsMaster.workerHeartbeat("192_1_1_2", metrics2);
-    assertEquals(11L, getGauge("metricA"));
+    waitForAndCheckResult(() -> (Long) getGauge("metricA") == 11L);
```

Comment:
why do only some of the asserts use this wait()? Is there a way to just wait for the update to happen, and then do the original asserts? It might be confusing to determine which assert or wait to call in which order.

Reply:
For now, I only use this wait() in the first assert check after worker's heartbeat(s). If one metric is updated, other metrics normally will also be updated.  I haven't come up with better ways to wait for the async metrics updater to finish. It will be better to use this waitAndCheck() function for all the assert checks to avoid the racing issues.